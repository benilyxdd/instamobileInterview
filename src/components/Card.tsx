import React from 'react';
import {
	View,
	Text,
	Image,
	StyleSheet,
	Dimensions,
	TouchableOpacity,
} from 'react-native';

type sizeType = 'sm' | 'md' | 'lg'; // small, medium, large -> string
export interface Props {
	imageUrl: string;
	title: string;
	subtitle: string;
	size: sizeType;
	onPress: () => void;
}

const screen = Dimensions.get('window');

const Card: React.FunctionComponent<Props> = ({
	imageUrl,
	title,
	subtitle,
	size,
	onPress,
}) => {
	return (
		<TouchableOpacity
			onPress={onPress}
			testID="card"
			style={{
				...styles.cardContainer,
				...styles[size],
			}}>
			<View style={styles.imageContainer}>
				<Image
					testID="image"
					style={styles.image}
					source={{ uri: imageUrl }}
				/>
			</View>
			<Text testID="title" style={styles.titleText}>
				{title}
			</Text>
			<Text testID="subtitle" style={styles.categoryText}>
				{subtitle}
			</Text>
		</TouchableOpacity>
	);
};

const styles = StyleSheet.create({
	sm: {
		width: screen.width * 0.4,
		height: screen.height * 0.3,
		marginHorizontal: screen.width * 0.05,
		marginVertical: screen.height * 0.025,
	},
	md: {
		width: screen.width * 0.9,
		height: screen.height * 0.33,
		marginHorizontal: screen.width * 0.05,
		marginVertical: screen.height * 0.025,
	},
	lg: {
		width: screen.width * 0.9,
		height: screen.height * 0.5,
	},

	cardContainer: {
		borderColor: 'gray',
		borderWidth: 1,
		borderRadius: 20,
		overflow: 'hidden',
	},
	categoryText: {
		textAlign: 'center',
	},
	image: {
		height: '100%',
		width: '100%',
	},
	imageContainer: {
		borderRadius: 20,
		height: '75%',
		width: '100%',
		overflow: 'hidden',
	},
	titleText: {
		fontWeight: 'bold',
		fontSize: 15,
		textAlign: 'center',
		paddingTop: '1%',
		paddingBottom: '2.5%',
	},

	test: {
		backgroundColor: 'red',
	},
});

export default Card;
