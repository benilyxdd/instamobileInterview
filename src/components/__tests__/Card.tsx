import React from 'react';
import { create } from 'react-test-renderer';

import Card from '../Card';

const card = create(<Card />);

it('match snapshot', () => {
	expect(card).toMatchSnapshot();
});
