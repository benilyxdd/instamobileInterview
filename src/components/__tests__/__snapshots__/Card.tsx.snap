// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card components snapshot 1`] = `
<View
  style={
    Object {
      "borderColor": "gray",
      "borderRadius": 20,
      "borderWidth": 1,
      "height": 440.22,
      "overflow": "hidden",
      "width": 675,
    }
  }
  testID="card"
>
  <View
    style={
      Object {
        "borderRadius": 20,
        "height": "70%",
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://www.texanerin.com/content/uploads/2019/06/nobake-chocolate-cookies-1-650x975.jpg",
        }
      }
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
      testID="image"
    />
  </View>
  <Text
    style={
      Object {
        "fontSize": 15,
        "fontWeight": "bold",
        "paddingBottom": "2.5%",
        "paddingTop": "1%",
        "textAlign": "center",
      }
    }
    testID="title"
  >
    Oatmeal Cookies
  </Text>
  <Text
    style={
      Object {
        "textAlign": "center",
      }
    }
    testID="category"
  >
    Cookies
  </Text>
</View>
`;
